Adding generated openAPIV3Schema from hack tool (not embedded)

matzew opened the pull request:
Signed-off-by: Matthias Wessendorf <mwessend@redhat.com>

Fixes #

<!-- Please include the 'why' behind your changes if no issue exists -->

## Proposed Changes

- schema for KafkaChannel
-
-

<!--
If this change has user-visible impact, follow the instructions below.
Examples include:

- 🎁 Add new feature
- 🐛 Fix bug
- 🧽 Update or clean up current behavior
- 🗑️ Remove feature or internal logic

Otherwise delete the rest of this template.
-->

**Release Note**

<!--
🗒️ If this change has user-visible impact, write a release note in the block
below. Include the string "action required" if additional action is required of
users switching to the new release, for example in case of a breaking change.

Write as if you are speaking to users, not other Knative contributors. If this
change has no user-visible impact, no release-note is needed.
-->

```release-note
adding openAPIV3Schema for KafkaChannel
```

**Docs**

<!--
📖 If this change has user-visible impact, link to an issue or PR in
https://github.com/knative/docs.
-->

matzew committed:
:broom: Adding generated openAPIV3Schema from hack tool (not embedded)

Signed-off-by: Matthias Wessendorf <mwessend@redhat.com>

matzew commented:
/assign @lionelvillard

lionelvillard commented on config/channel/resources/kafkachannel-crd.yaml:
```diff
@@ -31,12 +31,197 @@ spec:
+              type: object
+              properties:
+                numPartitions:
+                  description: NumPartitions is the number of partitions of a Kafka topic. By default, it is set to 1.
```
The `default` field is GA since 1.17. Maybe it makes sense to add it here for extra documentation purpose?

lionelvillard commented:
/hold if you want to experiment with `default`

matzew committed:
Adding default for replication and partion fields

Signed-off-by: Matthias Wessendorf <mwessend@redhat.com>

matzew commented:
@lionelvillard thanks! That was useful!

I've updated the PR and I added two `default` fields for the `numPartitions` and `replicationFactor` fields, as discussed/shown in here: https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/

/unhold
